docs/sc-memory/api/cpp/extended/permissions_api.md

```diff
@@ -0,0 +1,809 @@
+      <td>No, he doesn't.</td>
+      <td>No, he doesn't.</td>
+      <td>No, he doesn't.</td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>Yes. The sc-construction is in a sc-structure that indicates that certain permissions are required. Some users have permissions to this sc-structure.</td>
+      <td>Yes. He has global read permissions.</td>
+      <td>No, he doesn't.</td>
+      <td>No, he doesn't. The user hasn't local permissions for the specified sc-structure. But he can read any sc-constructions in knowledge base, which doesn't have permitted sc-structures.</td>
+    </tr>
+    <tr>
+      <td>5</td>
+      <td>Yes. The sc-construction is in a sc-structure that indicates that certain permissions are required. Some users have permissions to this sc-structure.</td>
+      <td>No, he doesn't.</td>
+      <td>Yes. He has local write permissions for the specified sc-structure.</td>
+      <td>Yes. The user can only add sc-elements into the specified sc-structure. But he can't handle any sc-constructions in knowledge base other than the given one.</td>
+    </tr>
+    <tr>
+      <td>6</td>
+      <td>Yes. The sc-construction is in a sc-structure that indicates that certain permissions are required. Some users have permissions to this sc-structure.</td>
+      <td>Yes. He has global write permissions.</td>
+      <td>Yes. He has local write permissions for the specified sc-structure.</td>
+      <td>Yes. The user can add sc-elements into the specified sc-structure. And he can add sc-element into any sc-constructions in knowledge base other than the given one.</td>
+    </tr>
+  </tbody>
+</table>
+
+---
+
+### **Examples of using user permissions**
+
+If you want to update permissions for a user, then you can describe them in the knowledge base in any format convenient for you (SCs-code or SCg-code) or do it programly.
+
+#### **Example for adding (updating) user permissions**
+
+If you want to add permissions for user constantly, specify permissions for this user in the knowledge base. See examples of local and global user permissions above.
+
+To do this programly you can create waiter to wait sc-event of adding outgoing sc-arc from permissions class and add permissions for this user. After waiting this sc-event, your user will have new user permissions.
+
+```cpp
+...
+// Find a user, for whom you need to add new permissions.
+ScAddr const & userAddr1 = context.HelperFindBySystemIdtf("user_1");
+// Find a structure, within which the user should have permissions.
+ScAddr const & structureAddr = context.HelperFindBySystemIdtf("my_structure");
+
+// Before adding new user permissions, specify that user doesn't have these
+// permissions. This way, you can subscribe to sc-event of erasing negative 
+// sc-arc, because when you add new permissions for user, all existing 
+// negative sc-arcs to permissions are erased. This logic will help you to 
+// wait for user's permissions to be changed.
```

Comment:
Such an abundance of comments looks like an attempt to tell something more from what was not told. This is not an example, this is a continuation of the theory. An example is needed. I talk about all content of this example and above

Reply:
Let's discuss